qiskit/transpiler/target.py

```diff
@@ -1066,59 +1067,79 @@ def from_configuration(
 
             qubit_properties = qubit_props_list_from_props(properties=backend_properties)
 
-        target = cls(
-            num_qubits=num_qubits,
-            dt=dt,
-            granularity=granularity,
-            min_length=min_length,
-            pulse_alignment=pulse_alignment,
-            acquire_alignment=acquire_alignment,
-            qubit_properties=qubit_properties,
-            concurrent_measurements=concurrent_measurements,
-        )
-        name_mapping = get_standard_gate_name_mapping()
-        if custom_name_mapping is not None:
-            name_mapping.update(custom_name_mapping)
-
+        if basis_gates is None:
+            # The Target class requires basis_gates.
+            # If there are no basis_gates, we can't build a proper Target instance,
+            # but we can generate a "pseudo" target that holds connectivity constraints
+            # and is compatible with our transpilation pipeline
+            if num_qubits is None and coupling_map is not None:
+                num_qubits = len(coupling_map.graph)
+            target = FakeTarget(
```

Comment:
Nitpicking alert: 
I'm not a big fan in general of the pattern that a class' constructor instantiates an object which is not of the class itself, even if it's a subclass. And here even more so since now we are basically allowing a user to create the internal `FakeTarget` via a user-facing API  when passing `basis_gates=None` to `Target.from_configuration`. Do we really want to support this use-case? I am not sure, probably not until at least `Target` fully and officially supports partial initialization. 

One way to alleviate this could be to have `generate_preset_pass_manager` the only factory of `FakeTarget` via say `FakeTarget.from_configuration`. This may require refactoring `Target.from_configuration` a bit, namely making the code which adds instructions to the newly instantiated target a class function of `Target` such that `FakeTarget` can also use it it's own `from_configuration` method. Though if only coupling map is supplied none of the code is actually used so maybe this type of refactoring is not even needed if we only want to support coupling-based only initialization of `FakeTarget`.

Reply:
The class was refactored in [e79f06c](https://github.com/Qiskit/qiskit/pull/12850/commits/e79f06c34cb8af3acef9ce90da6ee160a5ba1b2e), I agree that this wasn't the best pattern. Now there's a clear path for `Target` and `_FakeTarget` creation and the logic is left to the pass manager generation.